Collaborative writing section

agitter opened the pull request:
This section may need more feedback and revision than the others.  It is hard to find the right middle ground for describing the writing and review process.  Someone who is familiar with version control will find this boring and trivial.  Someone who isn't familiar may not understand the process or appeal after reading this.  That's why I recommend adding a graphic.

agitter committed:
Initial collaborative writing draft

dhimmel commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+GitHub's review interface allows anyone to comment on the changes, globally or at specific lines, asking questions or requesting modifications from the author.
+Once all requested modifications were made, the manuscript owners, a subset of authors with elevated GitHub permissions, could formally approve the pull request and merge the changes into the official master version.
+`TODO: need a figure with a flowchart showing this process`
+`TODO: could cite https://medium.com/towards-data-science/opportunities-and-obstacles-for-deep-learning-in-biology-and-medicine-6ec914fe18c2, the figure is nice`
```
Should cite [this post](https://medium.com/towards-data-science/opportunities-and-obstacles-for-deep-learning-in-biology-and-medicine-6ec914fe18c2) somewhere if not here.

agitter replied:
Yes, I'll add it as a URL citation during my next push.

dhimmel commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+A contributor could then add and revise any files, grouping these changes into _commits_.
+When the changes were ready to be reviewed, the series of commits could be submitted as a _pull request_ through GitHub, which notified other authors of the pending changes.
+GitHub's review interface allows anyone to comment on the changes, globally or at specific lines, asking questions or requesting modifications from the author.
+Once all requested modifications were made, the manuscript owners, a subset of authors with elevated GitHub permissions, could formally approve the pull request and merge the changes into the official master version.
```
I don't think "owners" is the right term. In GitHub parlance, those with "write" or "admin" access can merge PRs. In software, these individuals are generally called maintainers. GitHub uses the "Member" (or "Owner") label next to individuals names who fit this criteria.

agitter replied:
What would be the best phrasing?  `authors with write permissions could formally approve...`?

I'm finding it difficult to switch between lay terms and git/GitHub jargon.  I didn't bring up branches at all.

cgreene replied:
I like using the term "maintainers". It's software dev-y enough that it triggers "oh, this is different" but in the context of the sentence it gets explained shortly thereafter.

dhimmel commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+These platforms typically offer version control and other features to support multi-author editing.
+`TODO: some use git in the back end, is that relevant?`
+However, their critical limitation for open writing is that they give all authors permission to modify all parts of the manuscript.
+This makes it difficult to maintain consistent content and tone and is susceptible to malicious edits.
```
Suggesting edits on Google Docs fixes this issue. Maybe the difference here is that here we're writing the source? Where Google Docs is WYSIWYG. In addition, the underlying version control system of git is open and more versatile.

Addressing collaborative writing and manuscript writing in one paragraph is difficult... so we should reflect on what truly differentiates our approach.

agitter replied:
If Google Docs has that feature, I should de-emphasize or entirely remove that concern in this paragraph.

What differentiating features should we emphasize in this first paragraph?  Or what limitations of other approaches should we point out?  Credit attribution and the review process (if Google Doc doesn't offer anything similar) are important but covered in the next paragraphs.

cgreene replied:
Google docs + suggested edits are the way to go, but can't someone still just straight up edit? Is there a way to give someone only suggest permissions? The google docs docs [meta] are surprisingly unclear on the topic:
`Other people can make suggested changes when you give them permission to comment on or make changes to your file.`

cgreene replied:
I like the points in the paragraph below related to issues. The fact that they can be interleaved with discussions around changes reduces duplication of discussion.

dhimmel replied:
> can't someone still just straight up edit? Is there a way to give someone only suggest permissions?

If you share a Google Doc with permission level "Can comment", then others can suggest changes but not edit directly.

agitter replied:
> If you share a Google Doc with permission level "Can comment", then others can suggest changes but not edit directly.

This permission setting makes sense to me, but I don't have experience using it.  I created a test document for us to experiment with.

agitter replied:
There is a fairly detailed version history, but I don't see anything equivalent to the GitHub contributors page.  It would be hard to trace the contributions of individual authors in the Google Doc.

dhimmel commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+`TODO: could cite https://medium.com/towards-data-science/opportunities-and-obstacles-for-deep-learning-in-biology-and-medicine-6ec914fe18c2, the figure is nice`
+The entire process can be orchestrated through GitHub with a web browser if a contributor does not want to use a git client on their own computer.
+We found that this workflow was the best compromise between fully unrestricted editing and a more heavily-structured approach that limits the authors or the sections they can edit.
+In addition, authors are associated with their commits, making it easy for contributors to receive credit for their work.
```
Perhaps we could add some citations about credit problems in academia or ghostwriting, such as https://doi.org/10.1371/journal.pmed.1000023.

agitter replied:
Yes, it would be nice to have external support that this is a real problem.

cgreene commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+`TODO: survey other existing options, see if there are better references`
+These platforms typically offer version control and other features to support multi-author editing.
+`TODO: some use git in the back end, is that relevant?`
+However, their critical limitation for open writing is that they give all authors permission to modify all parts of the manuscript.
```
Other limitations: lack of _issue_ support and on them it may be potentially hard to separate contributions. But ultimately we probably could have done this on another platform, so maybe "critical" is too strong of a word.

cgreene commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+`TODO: need a figure with a flowchart showing this process`
+`TODO: could cite https://medium.com/towards-data-science/opportunities-and-obstacles-for-deep-learning-in-biology-and-medicine-6ec914fe18c2, the figure is nice`
+The entire process can be orchestrated through GitHub with a web browser if a contributor does not want to use a git client on their own computer.
+We found that this workflow was the best compromise between fully unrestricted editing and a more heavily-structured approach that limits the authors or the sections they can edit.
```
Well - technically we didn't test others. Maybe `was an effective compromise between...`?

cgreene reviewed (commented):
a few thoughts.

agitter committed:
Partial changes in response to dhimmel and cgreene comments

agitter committed:
Additional changes in response to dhimmel and cgreene comments

agitter commented:
@cgreene @dhimmel this is ready for another round of review.  I de-emphasized the editing permissions and focused more on issues and attribution.

cgreene reviewed (approved):
I like this version! 👍

dhimmel commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+
+There are many existing collaborative writing platforms ranging from rich text editors, which support Microsoft Word documents or similar formats, to LaTeX-based systems for more technical writing [@url:https://www.overleaf.com/ @url:https://www.authorea.com/].
+`TODO: survey other existing options, see if there are better references`
+These platforms typically offer version control, multiple permission levels, and other functionality to support multi-author document editing.
```
ideally may be more accurate than typically. Perhaps change the "and" to "or".

dhimmel commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+There are many existing collaborative writing platforms ranging from rich text editors, which support Microsoft Word documents or similar formats, to LaTeX-based systems for more technical writing [@url:https://www.overleaf.com/ @url:https://www.authorea.com/].
+`TODO: survey other existing options, see if there are better references`
+These platforms typically offer version control, multiple permission levels, and other functionality to support multi-author document editing.
+Although they work well for editing, they lack sufficient features for managing a collaborative manuscript and attributing precise credit, which are important for open writing.
```
Nice.

dhimmel commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+Issues also served as an open to-do list and a forum for debating the main message, themes, and topics of the review.
+
+GitHub and the underlying git version control system also structured the writing process.
+An official master version is _forked_ by individual contributors.
```
Does `master` here create confusion with the master branch? The forking is of the repository of record, and forks all the branches (I believe).

agitter replied:
Yes, that's confusing.  I was hiding the details about branching.  I'll refer to this as the `official` version instead.  I'll also reference https://doi.org/10.1371/journal.pcbi.1004668 for readers who want to learn more about the git terminology.

dhimmel commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+The entire process can be orchestrated through GitHub with a web browser if a contributor does not want to use a git client on their own computer.
+
+We found that this workflow was an effective compromise between fully unrestricted editing and a more heavily-structured approach that limits the authors or the sections they can edit.
+In addition, authors are associated with their commits, which makes it easy for contributors to receive credit for their work and prevents ghostwriting [@doi:10.1371/journal.pmed.1000023].
```
"helps prevent" --  :ghost: :writing_hand: is still possible. What's needed is also a repo policy that commits of others work must notify maintainer. (not necessary to discuss here)

dhimmel commented on content/02.main-text.md:
```diff
@@ -21,6 +21,34 @@ Although we requested that some authors participate for their specific expertise
+
+We found that this workflow was an effective compromise between fully unrestricted editing and a more heavily-structured approach that limits the authors or the sections they can edit.
+In addition, authors are associated with their commits, which makes it easy for contributors to receive credit for their work and prevents ghostwriting [@doi:10.1371/journal.pmed.1000023].
+The GitHub contributors page summarizes all edits and commits from each author, providing aggregated information that is not available on other collaborative writing platforms.
```
Important to note that the underlying version control system, git, is open... therefore there really is no limit to what contribution analyses could be performed retroscpectively. This is different from Google Docs, where your interaction with the version history is limited to a predefined interface and is not machine-readable.

agitter replied:
I'll work on a revision to note this.  I also want to be fair to other platforms that also have git/gitHub integration and phrase things accordingly.

Authorea [syncs to GitHub](https://intercom.help/authorea/user-settings/user-settings/syncing-articles-to-github) ([example](https://github.com/jdblischak/git-for-science)) and Overleaf has some [support for GitHub](https://www.overleaf.com/blog/195-new-collaborate-online-and-offline-with-overleaf-and-git-beta#.WYnKg4jythE) that I haven't looked at carefully.

agitter committed:
Include dhimmel suggestions, new references

agitter commented:
@dhimmel In 2f72290 I included your suggested phrasing, added a closing sentence about the git history, and added new references on collaborative writing platforms and git/GitHub.
